### PR TITLE
fix: notification error navigate

### LINKED ISSUE #2533: Click on the LU error notification does not show the LU error content
## Describe the bug

. Add a line of LU data with error to the LU trigger phrase
. Notice that an error icon showing besides the "Start Bot" button
. Click the button to get the detail LU error notifications
. Click each error notification
. Composer jump to the Lu All-up view

Expect:
Composer shows the LU data with error I added.

Result:
the error data didn't show up in the view.

## Browser

- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [ ] Windows
- [ ] Ubuntu

## Screenshots

. error data added
image
. error indicator
image
. error notifications
image
. when click the error notification item, composer jump to all-up view, and all data are correct there.
image

### LINKED ISSUE #2546: need the botid path for the dialog referenced in "Defined in" in lu all up
- click "Defined in" to navigate to the corresponding dialog in lu all up 
image
- hit following error
image
- This is due to no botid url attached 
image

### PR DESCRIPTION
**Pull request by zhixzhan**
## Task Item

closes #2533 
closes #2546 

## Screenshots

Untitled2

### CODE REVIEW

File: Composer/packages/client/src/pages/notifications/index.tsx [hunk 1]
```diff
@@ -24,17 +25,19 @@ const Notifications: React.FC<RouteComponentProps> = () => {
...
       }
       navigateTo(url);
     },
     [NotificationType.LU]: (item: INotification) => {
-      let uri = `/bot/${projectId}/language-understanding/${item.id}/edit#L=${item.diagnostic.range?.start.line || 0}`;
+      const dialogId = getBaseName(item.id);
+      let uri = `/bot/${projectId}/language-understanding/${dialogId}/edit#L=${item.diagnostic.range?.start.line || 0}`;
       if (item.dialogPath) {
-        uri = convertPathToUrl(item.id, item.dialogPath);
+        uri = convertPathToUrl(projectId, dialogId, item.dialogPath);
```
Comment (zhixzhan) [inline Composer/packages/client/src/pages/notifications/index.tsx:40]: @liweitian please help check line 40, did the params correctly at very first?

**Comment by cwhitten** [comment]
@zhixzhan why in the scenario of an inline LG error on a template like `bdfactivity-23orinfe` does the navigation take you to the all-up page. I thought the intention was to navigate to the action directly in this case of inline LG or LU errors. Was this behavior changed by design?

**Comment by zhixzhan** [comment]
@cwhitten By design it should navigate to inline, current not working come from 2 reason.
1. after involved `projectId` &`locale` the url calculation did not been updated (fixed now)
2. for some scenario of inline LU error ( maybe also LG in lower chance), our reverse calculate error's dialogPath rely on luParser can parse the error intent, so the error can be located in one intent, then the dialog use that this intent can be located, and calculate to a dialog path.

for 2 in conclusion,  currently, not every inline error can be navigate back to inline editor because of our lg/lu Parser ability.

**Comment by cwhitten** [comment]
> @cwhitten By design it should navigate to inline, current not working come from 2 reason.
> 
> 1. after involved `projectId` &`locale` the url calculation did not been updated (fixed now)

I just tried it out and didn't see any different in behavior with a LG error. It still navigated to the all-up page

**Comment by zhixzhan** [comment]
> > @cwhitten By design it should navigate to inline, current not working come from 2 reason.
> > 
> > 1. after involved `projectId` &`locale` the url calculation did not been updated (fixed now)
> 
> I just tried it out and didn't see any different in behavior with a LG error. It still navigated to the all-up page

can you paste the error lg file or bot, let me check again. if a LG error ( like just updated screenshot ) it did navigate to form editor

**Comment by zhixzhan** [comment]
@cwhitten updated, found another reason is lgNamePattern not updated from 'bdfactivity-233534' to 'bdfactivity-23orinfe'

**Pull request merged**